packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -457,6 +473,23 @@ function NativeStackViewInner({ state, navigation, descriptors }: Props) {
   );
 }
 
+const handleHeaderHeightChangeEvent = (
+  height: number,
+  cachedHeaderHeight: React.MutableRefObject<number>,
+  animatedHeaderHeight: Animated.Value
+) => {
+  if (cachedHeaderHeight.current !== height) {
+    // Currently, we're setting value by Animated#setValue, because we want to cache animated value.
+    // Unfortunately, because of the bug that exists on Fabric, where native event drivers
+    // for Animated objects are being created after the first notifications about the header height
+    // from the native side, it's not possible to use Animated#event, as
+    // we won't receive initial values on appearing screens.
+    // TODO(tboba): Replace animated#setValue to Animated#event.
```

Comment:
Is there a way to set it only initially and then rely on the animated.event? 
Please, attach a link, etc.

Reply:
Yeah it's possible (and currently we're [using](https://github.com/tboba/react-navigation/blob/%40tboba/react-native-screens-3.27.0/packages/native-stack/src/views/NativeStackView.native.tsx#L242) values from `useHeaderHeight` as initial value), but I think that we shouldn't really avoid this problem - this hook not only provide the values on the header height updates, but also guarantees that initial values are the real ones that comes from the native side - this could fix a potential problem that users are facing when there's a new iOS update that sometimes changes the status bar height (take a look on #11338, #11655 😄). I think this hook should provide as pure as possible experience of getting the values of header height dynamically.

Currently I know where the problem lies and I'll try to patch it in official react-native repo, but for now using the `setValue` should be sufficient (even if this works really flaky on Fabric). What do you think?

Also, thanks for spending the time on reviewing this PR! ❤️

Reply:
As mentioned earlier in the comment concluding my first review, I fail to see the motivation for this PR if we don't use Animated.event.

Particularly, now the user can simply do 

```
const animatedHeight = useAnimatedValue()
const  height = useHeaderHeight()
animatedHeight.setValue(height) // maybe check if equal with ref, then avoid update 
```

Reply:
Wouldn't it be possible even if we use Animated.event? I believe we have no influence on whether the user sets the value on its own or not 🤷. There's only a difference on updating the value during the `onHeaderHeightChange` event. The main downside of using `setValue` is that we cannot use animated driver on setting the value.

Reply:
I didn't mean I wanted to prevent the user from doing anything. My point is that the exposed hook functionality (`useAnimatedHeaderHeight`) proposed in this PR can be quickly achieved by the lines of code I pasted above. 

In other words, why do we need to add an extra hook for doing the functionality, which is already easily available right now?

Reply:
Oh, yeah I understand right now.
The main reason is that `useHeaderHeight` works by returning hardcoded values. This prevents the user to follow the header height updates resulted by, for example, using and swiping the large header on iOS. As I've mentioned above, it also guarantees that values returned by this hook will always be correct, no matter of status bar height changes from iOS, as it returns the values from the native side (check the example above how does it work in practice).

The main difference is that useHeaderHeight returns the static values by checking the conditions from JS, whereas useAnimatedHeaderHeight returns the values from the native side dynamically by receiving events. I agree that user can simply put the value from useHeaderHeight to useAnimatedValue, but it won't be updated dynamically, sadly.

Reply:
Okay, I see. It was me not understanding how it works.
In this case, I think, I'd prefer to make it an `Animated.event` anyway and accept that it won't work for now. 
It is okay as eventually it will be fixed in the RN core. Furthermore, we would merge it to the main branch and the 7.x still has a long enough backlog so the issue will be probably fixed. Let's leave a TODO there.

Reply:
Sure! I've replaced that in https://github.com/react-navigation/react-navigation/pull/11663/commits/d202582e238288c04f5185449a739211c345e1ec.